inc/usersim/ex.h

```diff
@@ -198,6 +198,80 @@ USERSIM_API _Ret_maybenull_ void*
 ExAllocatePoolWithTagCPP(
     _In_ __drv_strictTypeMatch(__drv_typeExpr) POOL_TYPE pool_type, SIZE_T number_of_bytes, ULONG tag);
 
+/**
+ * @brief Allocate memory.
+ * @param[in] pool_type Pool type to use.
+ * @param[in] size Size of memory to allocate.
+ * @param[in] tag Pool tag to use.
+ * @param[in] initialize False to return "uninitialized" memory.
+ * @returns Pointer to memory block allocated, or null on failure.
+ */
+__drv_allocatesMem(Mem) _Must_inspect_result_ _Ret_writes_maybenull_(size) void* usersim_allocate_with_tag(
```

Comment:
nit: what is the difference between `usersim_allocate_with_tag()` with PoolType `NonPagedPoolNxCacheAligned` and `usersim_allocate_cache_aligned()`?

Reply:
Haven't done cleanup, but usersim_allocate_cache_aligned doesn't do anything with the tag, so I think it should be removed.  usersim_allocate_cache_aligned_with_tag and usersim_allocate_with_tag() with PoolType NonPagedPoolNxCacheAligned are similar, but usersim_allocate_with_tag should really detect any cache-aligned pool type (it doesn't yet).  I suspect in the DLL form, nothing calls usersim_allocate_cache_aligned or usersim_allocate_cache_aligned_with_tag, but will check.

Reply:
I will clean this up in a separate PR as part of the work for https://github.com/microsoft/ebpf-for-windows/issues/2677.  In the meantime, I have added some TODO comments in the code.